tests/src/main/scala/scalaz/tests/ACatenable1Tests.scala

```diff
@@ -0,0 +1,147 @@
+package scalaz
+package tests
+
+import scala.Int
+import scala.Predef.{ Map, String }
+
+import data._
+import tc._
+import Scalaz._
+
+import laws._
+
+import testz._
+import z._
+
+import ACatenable1.lift
+
+object ACatenable1Tests {
+  // these type aliases unnerve me with their presence
+  type R[A, B] = ACatenable1[Map, A, B] => Map[A, B]
+  type L[A, B] = Map[A, B] => ACatenable1[Map, A, B]
+  val foldF: ACatenable1[Map, ?, ?] ~~> Map = Forall2.of[R](_.fold)
+  val liftF: Map ~~> ACatenable1[Map, ?, ?] = Forall2.of[L](ACatenable1.lift(_))
+
+  // specified in `compose`-order
+  val fst = Map(
+    (6, 9),
+    (7, 10),
+    (8, 11)
+  )
+  val snd = Map(
+    (3, 6),
+    (4, 7),
+    (5, 8),
+  )
+  val thd = Map(
+    (0, 3),
+    (1, 4),
+    (2, 5),
+  )
+
+  def const(s: String): ACatenable1[Biconst[String, ?, ?], Int, Int] = lift(Biconst(s))
+
+  def tests[T](harness: Harness[T], sequence: (T, T) => T): T = {
+    import harness._
+
+    sequence(
+      section("laws")(
+        section("lawful semicategory")(
+          test("associativity") { () =>
+            SemicategoryLaws.composeAssoc(fst, snd, thd)(assertEqualNonEmptyMaps)
+          },
+        ),
+        section("free semicategory")(
```

Comment:
This is the idea behind testing free structures: we test that `foldMap` is a homomorphism, and that `lift` is a homomorphism, when observed by `foldMap`.

Reply:
Strictly speaking you should also ensure that all observations are done through `foldMap`, right? That ensures that there is no extra fluff.

Reply:
Right, that's what I'm doing, isn't it?